File: Frontend/src/lib/settings/op-methods-catalog.ts
/** FA §12.1 OP (payment) methods — mirror Backend op_methods_catalog.py */

export type OpMethodDef = { id: string; label: string };

export const OP_METHOD_CATALOG: OpMethodDef[] = [
  { id: "cash", label: "Cash" },
  { id: "cheque", label: "Cheque" },
  { id: "pay_order", label: "Pay order" },
  { id: "demand_draft", label: "Demand draft" },
  { id: "bank_transfer", label: "Bank transfer / TT" },
  { id: "online", label: "Online payment" },
  { id: "credit_card", label: "Credit card" },
  { id: "pdc", label: "Post-dated cheque" },
];

export const DEFAULT_OP_METHOD_IDS = OP_METHOD_CATALOG.map((m) => m.id);

export function opMethodLabel(id: string): string {
  return OP_METHOD_CATALOG.find((m) => m.id === id)?.label ?? id;
}

File: Backend/src/app/constants/op_methods_catalog.py
"""FA §12.1 OP (payment) methods — defaults for receipts and bill payments."""

from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class OpMethodDef:
    id: str
    label: str


OP_METHOD_CATALOG: tuple[OpMethodDef, ...] = (
    OpMethodDef("cash", "Cash"),
    OpMethodDef("cheque", "Cheque"),
    OpMethodDef("pay_order", "Pay order"),
    OpMethodDef("demand_draft", "Demand draft"),
    OpMethodDef("bank_transfer", "Bank transfer / TT"),
    OpMethodDef("online", "Online payment"),
    OpMethodDef("credit_card", "Credit card"),
    OpMethodDef("pdc", "Post-dated cheque"),
)

DEFAULT_OP_METHOD_IDS: list[str] = [m.id for m in OP_METHOD_CATALOG]

DEFAULT_OP_METHODS_SETTINGS: dict[str, object] = {
    "defaultPaymentMethod": "cash",
    "enabledMethods": list(DEFAULT_OP_METHOD_IDS),
}
